code/modules/events/disease_outbreak.dm

```diff
@@ -1,41 +1,92 @@
-		D = A
+		chosen_disease = create_virus(severity * 2)	//Severity of the virus depends on the event severity level
+	chosen_disease.carrier = TRUE
 
-	D.carrier = TRUE
+/datum/event/disease_outbreak/start()
+	GLOB.current_pending_disease = chosen_disease
+	var/severity_text = "undefined (contact a coder)"
+	switch(severity)
+		if(EVENT_LEVEL_MUNDANE)
+			severity_text = "mundane"
+		if(EVENT_LEVEL_MODERATE)
+			severity_text = "moderate"
+		if(EVENT_LEVEL_MAJOR)
+			severity_text = "major"
 
-/datum/event/disease_outbreak/announce()
-	GLOB.event_announcement.Announce("Confirmed outbreak of level 7 major viral biohazard aboard [station_name()]. All personnel must contain the outbreak.", "Biohazard Alert", 'sound/AI/outbreak7.ogg')
 
-/datum/event/disease_outbreak/start()
-	for(var/mob/living/carbon/human/H in shuffle(GLOB.alive_mob_list))
-		if(!H.client)
-			continue
-		if(issmall(H)) //don't infect monkies; that's a waste
-			continue
-		var/turf/T = get_turf(H)
-		if(!T)
-			continue
-		if(!is_station_level(T.z))
-			continue
+	for(var/mob/M as anything in GLOB.dead_mob_list) //Announce outbreak to dchat
+		if(istype(chosen_disease, /datum/disease/advance))
+			var/datum/disease/advance/temp_disease = chosen_disease.Copy()
+			to_chat(M, "<span class='deadsay'><b>Disease outbreak:</b> The next new arrival is a carrier of [temp_disease.name]! A [severity_text] disease with the following symptoms: [english_list(temp_disease.symptoms)]</span>")
+		else
+			to_chat(M, "<span class='deadsay'><b>Disease outbreak:</b> The next new arrival is a carrier of a [severity_text] disease: [chosen_disease.name]!</span>")
+
+//Creates a virus with a harmful effect, guaranteed to be spreadable by contact or airborne
+/datum/event/disease_outbreak/proc/create_virus(max_severity = 6)
+	var/datum/disease/advance/A = new /datum/disease/advance
+	A.symptoms = A.GenerateSymptomsBySeverity(max_severity - 1, max_severity, 2) //Choose "Payload" symptoms
+	A.AssignProperties(A.GenerateProperties())
+	var/list/symptoms_to_try = transmissable_symptoms.Copy()
+	while(length(symptoms_to_try))
+		if(A.spread_text != "Blood")
+			break
+		if(length(A.symptoms) < VIRUS_SYMPTOM_LIMIT)	//Ensure the virus is spreadable by adding symptoms that boost transmission
+			var/datum/symptom/TS = pick_n_take(symptoms_to_try)
+			A.AddSymptom(new TS)
+		else
+			popleft(A.symptoms)	//We have a full symptom list but are still not transmittable. Try removing one of the "payloads"
 
-		if(!H.ForceContractDisease(D))
+		A.AssignProperties(A.GenerateProperties())
+	A.name = pick(GLOB.alphabet_uppercase) + num2text(rand(1,9)) + pick(GLOB.alphabet_uppercase) + num2text(rand(1,9)) + pick("v", "V", "-" + num2text(GLOB.game_year), "")
+	A.Refresh()
+	return A
+
+/datum/event/disease_outbreak/proc/populate_diseases()
+	for(var/candidate in subtypesof(/datum/disease))
+		var/datum/disease/CD = candidate
+		if(candidate in disease_blacklist)
```

Comment:
This doesn't consider subtypes of the datums in the blacklist, so it currently isn't working for example with the transform diseases.

Reply:
Fixed, thanks for reporting.